### PR TITLE
[MSKINS-245] Add code highlighting class to every <pre><code> combination

### PR DESCRIPTION
**Pull request by kwin**
Don't rely on custom div classes surrounding those as those are not always set.

### CODE REVIEW

**Review by michael-o — CHANGES_REQUESTED** [review]
This is broken as it does not reflect the skin structure AND the CSS anymore.

**Comment by kwin** [comment]
This is therefore only a draft to outline how this should work IMHO on the Skin side. Would you mind commenting on the general approach to only rely on `<pre><code>` for block codes?

**Comment by kwin** [comment]
This also fixes the missing indentation for `<div class="source">` (broken in both Fluido 1.x and 2.x), which e.g. can be seen in <https://jackrabbit.apache.org/filevault-package-maven-plugin/index.html>

Screenshot 2024-03-15 at 17 08 35

**Review by michael-o — APPROVED** [review]
Althrough this looks good and right, it needs to be remain a draft until the fixed Doxia (Sitetools) version is released this can be bumped:
https://github.com/apache/maven-fluido-skin/blob/d5eef2752bdb2e3d395438a502ab87ecbebd9e8c/src/main/resources/META-INF/maven/skin.xml#L25

**Comment by kwin** [comment]
Still requires a m-site-p version depending on m-doxia-sitetools M18+...

File: src/main/resources/META-INF/maven/site.vm [hunk 1]
```diff
@@ -239,7 +239,7 @@
 #*        *##set( $sourceStyle = "prettyprint" )
 #*      *##end
 ##
-#*      *#$bodyContent.replaceAll( '<div class="verbatim source">(\r?\n)?<pre>', '<div class="verbatim source"><pre class="' + $sourceStyle + '">' ).replaceAll( 'class="bodyTable"', 'class="table table-striped"' ).replaceAll( 'class="bodyTable bodyTableBorder"', 'class="table table-bordered table-striped"' )
+#*      *#$bodyContent.replaceAll( '<pre><code>', '<pre class="' + $sourceStyle + '"><code>' ).replaceAll( 'class="bodyTable"', 'class="table table-striped"' ).replaceAll( 'class="bodyTable bodyTableBorder"', 'class="table table-bordered table-striped"' )
```
Comment (michael-o) [inline src/main/resources/META-INF/maven/site.vm:242]: Since we don't need regex anymore, we can resort to just `replace()` here.

**Review by michael-o — CHANGES_REQUESTED** [review]

**Comment by michael-o** [comment]
This can now reduced to:
```patch
diff --git a/src/main/resources/META-INF/maven/site.vm b/src/main/resources/META-INF/maven/site.vm
index 5e0604f..26bb876 100644
--- a/src/main/resources/META-INF/maven/site.vm
+++ b/src/main/resources/META-INF/maven/site.vm
@@ -242 +242 @@
-#*      *#$bodyContent.replaceAll( '<div class="verbatim source">(\r?\n)?<pre>', '<div class="verbatim source"><pre class="' + $sourceStyle + '">' ).replaceAll( 'class="bodyTable"', 'class="table table-striped"' ).replaceAll( 'class="bodyTable bodyTableBorder"', 'class="table table-bordered table-striped"' )
+#*      *#$bodyContent.replace( '<pre><code>', '<pre class="' + $sourceStyle + '"><code>' ).replace( 'class="bodyTable"', 'class="table table-striped"' ).replace( 'class="bodyTable bodyTableBorder"', 'class="table table-bordered table-striped"' )
diff --git a/src/main/resources/css/maven-base.css b/src/main/resources/css/maven-base.css
index 829ef71..5747cd6 100644
--- a/src/main/resources/css/maven-base.css
+++ b/src/main/resources/css/maven-base.css
@@ -33 +33 @@ section > table.table,
-section > div.verbatim {
+section > pre {
```

**Comment by michael-o** [comment]
@kwin Should I rebase and merge this? I'd like to start the release.

**Comment by kwin** [comment]
> Should I rebase and merge this? I'd like to start the release

Yes, please.

**Review by michael-o — APPROVED** [review]
Will merge as soon as CI completes.

**Comment by jira-importer** [comment]
Resolve #259

**Pull request merged**